server/store/subscription_store.go

```diff
@@ -76,3 +84,30 @@ func (s *pluginStore) DeleteUserSubscription(user *User, subscriptionID string)
 	}).Debugf("store: deleted mattermost user subscription.")
 	return nil
 }
+
+func (s *pluginStore) AtomicModifySubscription(subscriptionID string, modify func(sub *Subscription) (*Subscription, error)) error {
+	return kvstore.AtomicModify(s.subscriptionKV, subscriptionID, func(initial []byte, storeErr error) ([]byte, error) {
+		if storeErr != nil {
+			return initial, storeErr
+		}
+
+		var sub *Subscription
+		if len(initial) > 0 {
+			err := json.Unmarshal(initial, sub)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		updated, err := modify(sub)
```

Comment:
If len(initial) == 0, here sub is null. Is this the expected behaviour?

Reply:
If `sub == nil` then the error at `line 90` would have returned already because of not finding it in the store. The `if len(initial) > 0` is a bit redundant. What do you suggest here, an additional check for `nil` just in case?

Reply:
If I understand correctly, len(initial) should be always >0 because if not, it would return a storeErr, right? If that is the case, maybe you can drop the if statement, and just leave a comment like `initial will always have data`. What do you think?